### ISSUE TITLE
[java] UnusedAssignment false positive in record compact constructor 

### ISSUE THREAD

**Issue by emouty** [pmd/pmd#5592]
**Affects PMD Version:**  7.11.0

**Rule:** [UnusedAssignment](https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#unusedassignment)

**Description:**

**Code Sample demonstrating the issue:**

```java
record TestRecord(Set<Integer> foo) {
    TestRecord {
        foo = foo == null ? emptySet() : new TreeSet<>(foo); // <!--- violation here
    }
}
```

**Expected outcome:**

PMD should not report false-positive about unused foo in compact constructor.

**Running PMD through:** *[Maven | Other]*
